gitbot/lib.py

```diff
@@ -157,7 +165,10 @@ def bump_version(branch, ref_sha, author=None, url=GETSENTRY_REPO_URL, dry_run=D
         except CommandError:
             run(f"git pull --rebase origin {branch}", cwd=repo_root)
 
+    if delete_temp_checkout:
+        os.rmdir(repo_root)
```

Comment:
heh well now it's not exception safe :(

here's two approaches that'd work for this:

```python
def bump_version(
    ...,
    temp_checkout: str | None = None,
):
    with contextlib.ExitStack() as ctx:
        if temp_checkout is not None:
            repo_root = temp_checkout
        else:
            repo_root = ctx.enter_context(tempfile.TemporaryDirectory)

        ...
```

the other alternative is make `temp_checkout` a required parameter (this is actually cleaner imo since it wouldn't need test-only code in the logic)